test/functorch/test_control_flow.py

```diff
@@ -73,6 +75,124 @@ def f(x, pred, pred2):
 
         self.assertEqual(result_false_true, torch.cos(x))
 
+    def test_cond_functionalized(self):
+        def true_fn(x):
+            y = x.sin()
+            y.add_(4)
+            return x.sin().max() + y.sum()
+
+        def false_fn(x):
+            return x.cos().min()
+
+        def f(x):
+            pred = x.shape[0] == 1
+            return cond(pred, true_fn, false_fn, [x])
+
+        example_inputs = (torch.ones(4, 5),)
+        functional_f = functionalize(f)
+        self.assertEqual(functional_f(*example_inputs), f(*example_inputs))
+
+        graph_module = make_fx(functionalize(f))(*example_inputs)
+        self.assertTrue(torch.allclose(graph_module(*example_inputs), f(*example_inputs)))
```

Comment:
nit: self.assertTrue(torch.allclose -> self.assertEqual, it works better with pytorch testing infra. Ditto for the other callsites below